connectors/services/job_cleanup.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""
+A task periodically clean up orphaned and stuck jobs.
+"""
+
+from connectors.byoc import ConnectorIndex, SyncJobIndex
+from connectors.logger import logger
+from connectors.services.base import BaseService
+
+STUCK_JOB_ERROR = "The job has not seen any update for some time."
+
+
+class JobCleanUpService(BaseService):
+    def __init__(self, config):
+        super().__init__(config)
+        self.idling = int(self.service_config.get("job_cleanup_interval", 60 * 5))
+        self.native_service_types = self.config.get("native_service_types", [])
+        if "connector_id" in self.config:
+            self.connectors_ids = [self.config.get("connector_id")]
+        else:
+            self.connectors_ids = []
+        self.connector_index = None
+        self.sync_job_index = None
+
+    async def _run(self):
+        logger.info("Successfully started Job cleanup task...")
+        self.connector_index = ConnectorIndex(self.es_config)
+        self.sync_job_index = SyncJobIndex(self.es_config)
+
+        try:
+            while self.running:
+                await self._process_orphaned_jobs()
+                await self._process_stuck_jobs()
+                await self._sleeps.sleep(self.idling)
+        finally:
+            if self.connector_index is not None:
+                self.connector_index.stop_waiting()
+                await self.connector_index.close()
+            if self.sync_job_index is not None:
+                self.sync_job_index.stop_waiting()
+                await self.sync_job_index.close()
+        return 0
+
+    async def _process_orphaned_jobs(self):
+        try:
+            logger.info("Start cleaning up orphaned jobs...")
+            connector_ids = [
+                connector.id
+                async for connector in self.connector_index.all_connectors()
+            ]
```

Comment:
Do we care about connector_ids here? I think all the data should be accessible from jobs, and we can pick up all orphaned jobs, does not matter which connector ids.

Reply:
We do need the `connector_ids`, as we need to find all jobs whose connector id does not exist.